client/client.go

```diff
@@ -364,57 +359,56 @@ func (c *client) Close() {
 }
 
 func (c *client) setServiceMode(newMode pdpb.ServiceMode) {
-	c.svcModeMutex.Lock()
```

Comment:
I prefer to keep this lock. Originally, I used the corresponding read lock in the getTSOClient, but eventually I used atomic.Value() to store tsoClient and removed the read lock. After that, I still decided to keep this lock and critical section in the setServiceMode() for the following reasons:
1. Have plan to improve error handling. After service mode switched at the server side, the requests will fail and the client needs to call serviceModeUpdate() directly and immediately instead of waiting serviceModeUpdateInterval 3s (I even changed the interval from the original 10s to the current value 3s) for this purpose. When the leader/primary being switched on the server side, we had the similar issue -- for this case, today we fail the requests immediately without retries though we schedule membership check immediately. All these need improvement.
2. This lock's cost can be overlooked (invoked every 3s) and it's inconvenient to add it back.
3. I prefer to move client.setServiceMode to serviceModeKeeper.setServiceMode so that it can be more explicit for changing serviceMode, tsoClient and tsoSvcDiscovery atomically, thus it's more modular.

Reply:
I added the lock back. PTAL.

Reply:
Sounds good. The change looks good to me.